debezium-server/debezium-server-http/src/main/java/io/debezium/server/http/HttpChangeConsumer.java

```diff
@@ -0,0 +1,126 @@
+    @PostConstruct
+    void connect() {
+        final Config config = ConfigProvider.getConfig();
+        String sink = System.getenv("K_SINK");
+        timeoutDuration = Duration.ofMillis(HTTP_TIMEOUT);
+
+        if (sink != null) {
+            sinkUrl = sink;
+        }
+        else {
+            sinkUrl = config.getValue(PROP_PREFIX + PROP_WEBHOOK_URL, String.class);
+        }
+
+        config.getOptionalValue(PROP_PREFIX + PROP_CLIENT_TIMEOUT, String.class)
+                .ifPresent(t -> timeoutDuration = Duration.ofMillis(Long.parseLong(t)));
+
+        LOGGER.info("Using sink URL: {}", sinkUrl);
+    }
+
+    private String getContentType(boolean cloudevent) {
+        if (cloudevent) {
+            return "application/cloudevents+json";
+        }
+        else {
+            return "application/json";
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, DebeziumEngine.RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+        HttpClient client = HttpClient.newHttpClient();
+
+        for (ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+
+            try {
+                HttpRequest.Builder requestBuilder = HttpRequest.newBuilder(new URI(sinkUrl))
+                        .timeout(timeoutDuration);
+
+                String value = "";
+                if (record.value() != null) {
+                    value = (String) record.value();
+
+                    // Since the ChangeEvent has already been serialized at this point, we need to inspect it to see if this is
+                    // a cloudevent, and set the correct content-type header.
+                    requestBuilder.setHeader("content-type", getContentType(value.contains("\"specversion\":\"1.0\"")));
+
+                    HttpRequest request = requestBuilder.POST(HttpRequest.BodyPublishers.ofString(value)).build();
+                    client.sendAsync(request, HttpResponse.BodyHandlers.ofString())
+                            .thenAccept(r -> {
+
+                                if ((r.statusCode() == HTTP_OK) || (r.statusCode() == HTTP_NO_CONTENT)) {
+                                    try {
+                                        committer.markProcessed(record);
+                                    }
+                                    catch (InterruptedException e) {
+                                        LOGGER.info("exception while committing work: " + e.toString());
+                                    }
+                                }
+                                else {
+                                    LOGGER.info("failed to publish event: " + r.body());
+                                }
+                            });
+                }
+
+            }
+            catch (Exception e) {
+                LOGGER.info("exception: " + e.toString());
```

Comment:
Let's better propagate the exception, i.e. stop the application in case of an exception.